apps/weatherclock/weatherclock.star

```diff
@@ -0,0 +1,568 @@
+"""
+Applet: Weather Clock
+Summary: Time and weather
+Description: Display current time and weather.
+Author: J. Keybl
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+
+DEBUG = False
+TIME_FONT = "6x13"
+WEATHER_FONT = "CG-pixel-3x5-mono"
+TIME_NIGHT_COLOR = "#333333"
+TEMPS_COLOR = "#FFFFFF"
+HUMIDITY_COLOR = "#0000FF"
+TIME_FORMAT_SEPARATOR = "3:04 PM"
+TIME_FORMAT_NO_SEPARATOR = "3 04 PM"
+TTL_SECONDS = 60
+DEFAULT_LOCATION = {
+    "lat": 38.9586,
+    "lng": -77.3560,
+    "locality": "Washington, D.C.",
+    "timezone": "America/New_York",
+}
+DEFAULT_API = {
+    "appID": "",
+}
+
+def main(config):
+    dt = 0
+    sunrise = 0
+    sunset = 0
+    dn = True
+    icon = ""
+    sTemps = ""
+    sHumidity = ""
+
+    location = config.get("location")
+    loc = json.decode(location) if location else json.decode(str(DEFAULT_LOCATION))
+    timezone = loc["timezone"]
+    lat = loc["lat"]
+    lon = loc["lng"]
+
+    weatherAPI = config.get("weatherAPI")
+    if weatherAPI == None:
+        api = json.decode(str(DEFAULT_API))
+        weatherAPI = api["appID"]
+
+    nightMode = config.bool("night_mode", False)
+    if nightMode == False:
+        dayModeSec = 0
+        nightModeSec = 1439
+    else:
+        nightModeStr = config.get("nightModeStart")
+        if nightModeStr == None:
+            nightModeStartHr = 23
+            nightModeStartMin = 0
+        else:
+            nightModeStartHr = int(nightModeStr[0:2])
+            if nightModeStartHr >= 24:
+                nightModeStartHr = 0
+            nightModeStartMin = int(nightModeStr[2:4])
+        dayModeStr = config.get("nightModeEnd")
+        if nightModeStr == None:
+            dayModeEndHr = 7
+            dayModeEndMin = 0
+        else:
+            dayModeEndHr = int(dayModeStr[0:2])
+            dayModeEndMin = int(dayModeStr[2:4])
+        nightModeSec = nightModeStartHr * 60 + nightModeStartMin
+        dayModeSec = dayModeEndHr * 60 + dayModeEndMin
+
+    openWeatherURL = "https://api.openweathermap.org/data/2.5/weather?lat=" + str(lat) + "&lon=" + str(lon) + "&APPID=" + str(weatherAPI) + "&units=imperial"
+
+    now = time.now().in_location(timezone)
+    ss = now.hour * 60 + now.minute
+    if DEBUG:
+        print(now)
+        print(now.format(TIME_FORMAT_SEPARATOR))
+
+    if ss >= dayModeSec and ss < nightModeSec:
+        sunrise_cached = cache.get("sunrise")
```

Comment:
The cache is scoped to the app, but not to individual users of the app. That means all users of the app will be accessing the same cached data here, which is probably not the behavior you're looking for. =)

I think a good path around it is to try to rely primarily on the http client's build in cache functionality. Read more abou that [over here](https://discuss.tidbyt.com/t/http-caching-client/5129).

Reply:
I have made the appropriate changes to the code to use the new the new HTTP Caching Client